src/Microsoft.Health.Fhir.CosmosDb/Features/Health/CosmosHealthCheck.cs

```diff
@@ -66,6 +67,26 @@ public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context
             const int maxExecutionTimeInSeconds = 30;
             const int maxNumberAttempts = 3;
             int attempt = 0;
+
+            // CosmosOperationCanceledException are "safe to retry on and can be treated as timeouts from the retrying perspective.".
+            // Reference: https://learn.microsoft.com/azure/cosmos-db/nosql/troubleshoot-dotnet-sdk-request-timeout?tabs=cpu-new
+            // Cosmos 503 and 449 are transient errors that can be retried.
+            // Reference: https://learn.microsoft.com/azure/cosmos-db/nosql/conceptual-resilient-sdk-applications#should-my-application-retry-on-errors
+            static bool IsRetryableException(Exception ex) =>
+                ex is CosmosOperationCanceledException ||
+                (ex is CosmosException cex && (cex.StatusCode == HttpStatusCode.ServiceUnavailable || cex.StatusCode == (HttpStatusCode)449));
+
+            void LogAdditionalRetryableExceptionDetails(Exception exception)
+            {
+                if (exception is CosmosException cosmosException && cosmosException.StatusCode == HttpStatusCode.ServiceUnavailable)
```

Comment:
Should we also add the logginc for HTTP449 and HTTP408?

Reply:
Actually this can be removed because it's added when useful to the exception message in the CosmosSDK.

https://github.com/Azure/azure-cosmos-dotnet-v3/blob/b4a4ac0f261b2e51ed5eb2b3e43fda946b36a771/Microsoft.Azure.Cosmos/src/Resource/CosmosExceptions/CosmosException.cs#L217